address/address.go

```diff
@@ -326,6 +330,7 @@ func (a *Tap) DecodeRecords() []tlv.Record {
 		newAddressInternalKeyRecord(&a.InternalKey),
 		newAddressTapscriptSiblingRecord(&a.TapscriptSibling),
 		newAddressAmountRecord(&a.Amount),
+		newProofCourierAddrRecord(&a.ProofCourierAddr),
```

Comment:
Needs to be added to the encode records as well. And to the unit test (which would've found this).